pulsar-client-go/pulsar/c_consumer.go

```diff
@@ -109,6 +112,24 @@ func subscribeAsync(client *client, options ConsumerOptions, callback func(Consu
 		C.pulsar_consumer_set_subscription_initial_position(conf, C.initial_position(options.SubscriptionInitPos))
 	}
 
+	if schema != nil {
+		if schema.GetSchemaInfo().Type != NONE {
+			cName := C.CString(schema.GetSchemaInfo().Name)
+			cSchema := C.CString(schema.GetSchemaInfo().Schema)
+			defer C.free(unsafe.Pointer(cName))
+			defer C.free(unsafe.Pointer(cSchema))
+			C.pulsar_consumer_configuration_set_schema_type(conf, C.pulsar_schema_type(schema.GetSchemaInfo().Type),
```

Comment:
following the comment I had above, if you have a method `pulsar_consumer_configuration_set_schema_info`, you can just convert a go `SchemaInfo` to a cpp `SchemaInfo`, correct? In this way, the properties will not be lost. Also in future, if we introduce new fields to `SchemaInfo`, they can be handled correctly in one place, without touching producer and consumer configuration in cpp again.

Reply:
yes, this is a good idea, but just an idea. in `SchemaInfo` struct, we may define a variety of types, such as: map. In c language, don't support `map`, of course, we can implement a map by ourselves or use `std::map` to implement it. But the biggest question is: how do we pass the map type of go to c? 

Go's built-in map is implemented in runtime. The implementation details are not exported. Even if it is exported, its memory is managed by GC. How to ensure that c code is used, it is not recycled.

Reply:
yes, this is a good idea, but just an idea. in `SchemaInfo` struct, we may define a variety of types, such as: map. In c language, don't support `map`, of course, we can implement a map by ourselves or use `std::map` to implement it. But the biggest question is: how do we pass the map type of go to c? 

Go's built-in map is implemented in runtime. The implementation details are not exported. Even if it is exported, its memory is managed by GC. How to ensure that c code is used, it is not recycled.

So far, I have not thought of a good way to deal with this problem.